### ISSUE TITLE
Che 7.0.0 Endgame Plan

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#13637]
# Che 7.0.0 Endgame

Target is the 14th of August (this may change)

## RULES

1. We release Che 7.0.0 as soon as:
**ENDGAME BUGS ARE FIXED**
**ENDGAME ENHANCEMENTS ARE COMPLETED**
**ENDGAME DOCUMENTATION IS AVAILABLE ON THE DOC SITE**
**E2E HAPPY PATH TEST IS GREEN (STEPS 1, 3 and 4)**
**CHE QE TEAM VERIFIES THAT THE RELEASE ACCEPTANCE CRITERIA ARE SATISFIED**
2. Nothing should be merged to master if it's not in the endgame plan (che-x, che-theia, chectl)
3. Nothing should be added to the endgame plan if not approved by the endgame doc owners: @nickboldt, @l0rd, @slemeur, @tsmaeder and @rhopp

### Bugs

- [x] Verify and fix devfile registry images to work on openshift  @ibuziuk
   - [x] https://github.com/redhat-developer/rh-che/issues/1434
   - [x] https://github.com/redhat-developer/rh-che/issues/1347  
   - [x] https://github.com/eclipse/che/issues/13454 
   - [x] https://github.com/eclipse/che/issues/13608
   - [x] https://github.com/eclipse/che/issues/13594
- [x] Verify and fix stacks language servers and debug adapters @tsmaeder
- [x] Stabilize CI for happy path scenario https://github.com/eclipse/che/issues/13523 @rhopp
- [x] Impossible to get a terminal in a workspace https://github.com/eclipse/che/issues/13584 @evidolob
- [x] Theia doesn't work with self signed certs https://github.com/eclipse/che/issues/13574 @evidolob
- [x] Issues with Theia configured and recent tasks https://github.com/theia-ide/theia/issues/5064 https://github.com/theia-ide/theia/issues/5067 https://github.com/theia-ide/theia/issues/4928 @vparfonov
- [x] chectl server:start cheimage parameter has no effect on multiuser minikube https://github.com/che-incubator/chectl/issues/161 @benoitf
- [x] chectl server:start fails if Che helm chart has status DELETE or DELETING https://github.com/che-incubator/chectl/issues/18 @benoitf
- [x] chectl uses deprecated method for creating workspaces from devfile https://github.com/eclipse/che/issues/13623 @l0rd
- [x] Clarify names of default stacks https://github.com/eclipse/che/issues/13592 @skabashnyuk
- [x] Preview URL not displayed while accessible https://github.com/eclipse/che/issues/13357 @evidolob
- [x] Unable to set `JAVA_OPTS` or `MAVEN_OPTS` in kubernetes component of devfile https://github.com/eclipse/che/issues/13340 @skabashnyuk
- [x] Some Projects don't work with Java 11 based Java plugin https://github.com/eclipse/che/issues/13547 @tsmaeder
- [x] Remove UD "Diagnostics" because it's not working https://github.com/eclipse/che/issues/13525 @ashumilova
- [x] Che Plugins panel is broken on minishift https://github.com/eclipse/che-theia/issues/296 @evidolob
- [x] Login using OpenShift 4.x OAuth doesn't work https://github.com/eclipse/che/issues/13659 @davidfestal 
- [x] Applications started with a che-commands (task) stop running on OpenShift https://github.com/eclipse/che/issues/13620 @vparfonov
- [x] Multi-user Che fails to start on minikube using helm https://github.com/eclipse/che/issues/13625 @skabashnyuk  
- [x] keycloak image init containers shouldn't do post-install stuff https://github.com/eclipse/che/issues/13870 @skabashnyuk
- [x] SSH plugin doesn't show if generation has been successful or not https://github.com/eclipse/che/issues/13857 @vparfonov
- [x] chectl workspace:start fails for multi-user Che https://github.com/eclipse/che/issues/13828 @benoitf
- [x] Go sample `permission denied` issues with dependencies https://github.com/eclipse/che/issues/13823 @tolusha
- [x] `Annotation key is too long` error when starting workspace, created from devfile https://github.com/eclipse/che/issues/13303 @skabashnyuk
- [x] Tasks and Che Commands should not use "machineName" https://github.com/eclipse/che/issues/13667 @vparfonov
- [x] GitHub identity is persisted locally after log-out https://github.com/eclipse/che/issues/13694 @ashumilova
- [x] Can't update Che 7 in Minishift to use a newer devfile registry https://github.com/eclipse/che/issues/13676 @evidolob 
- [x] configmaps "sshconfigmap" already exists https://github.com/eclipse/che/issues/13873 @vparfonov 
- [x] Outputs are not avialable when using theia tasks https://github.com/eclipse/che/issues/13876 @evidolob 

### Enhancements

- [x] New devfile based stacks should contain samples https://github.com/eclipse/che/issues/13529 @skabashnyuk 
- [x] GitHub PRs extension https://github.com/eclipse/che/issues/11867 @vparfonov
- [x] Git + SSH (generate with plugin + mount as secret) https://github.com/eclipse/che/issues/13494 https://github.com/eclipse/che/issues/13866 https://github.com/eclipse/che/issues/13857 @vparfonov
- [x] Deprecate storing of workspaces converted from devfiles https://github.com/eclipse/che/issues/13423 @skabashnyuk
- [x] Legacy environment variables cleanup https://github.com/eclipse/che/issues/12679 @skabashnyuk
- [x] chectl should use new CRD based operator https://github.com/che-incubator/chectl/issues/92 @l0rd
- [x] Che registries (plugin+devfile) should be part of Che deployment https://github.com/eclipse/che/issues/13633 @skabashnyuk ([helm]
- [x] Che registries (plugin+devfile) should be part of Che deployment (https://github.com/eclipse/che/issues/13557)) @davidfestal (operator)
- [x] Improve CI for devfile registry https://github.com/eclipse/che/issues/13837 https://github.com/eclipse/che/issues/13813 @amisevsk
- [x] Update release process to include new repositories (theia, broker, registries, chectl, operator) @vparfonov
- [x] Implement automated "happy path" e2e test (step 1, 3 and 4) https://github.com/eclipse/che/issues/12728 @rhopp
- [x] Use "happy path" e2e test as PR check on: che and che-theia #13326 @rhopp
- [x] Update projects widget in Create New Workspace Panel https://github.com/eclipse/che/issues/13665 https://github.com/eclipse/che/issues/13856 @ashumilova 
- [x] Remove Che CLI container images from build https://github.com/eclipse/che/issues/13649 @benoitf
- [x] Remove GWT @skabashnyuk @vparfonov
- [x] Remove Che CLI container images `eclipse/che:nightly` and `eclipse/che:latest` from DockerHub (has been removed from source code) @vparfonov
- [x] Deprecate and remove method /api/devfile/method https://github.com/eclipse/che/issues/13867 @skabashnyuk 
- [x] Give more flexibility on where devfile registry can served up https://github.com/eclipse/che/issues/13634
- [x] Adapt "Stacks" in UD to Che 7 https://github.com/eclipse/che/issues/13811 @ashumilova
- [x] Release script for che-operator https://github.com/eclipse/che/issues/13780 @davidfestal
- [x] Create a nightly image for che-theia image on branch 7.0.0 https://github.com/eclipse/che/issues/13782 @evidolob

### Docs

- The document draft should be shared on a google doc.
- Ping @l0rd, @slemeur and @rkratky once the article is ready for review (e.g. with a comment on the google doc).
- Review of the documents happens on the google doc
- Once the review is done the article will be published by the doc team to https://github.com/eclipse/che-docs/

#### Topics

- [x] Running Che remotely
    - [x] Installing Che on AWS, OpenShift 3, OpenShift 4 - **@benoitf** - [PR #768](https://github.com/eclipse/che-docs/pull/768)
    - [x] Installing Che on GCP - **@benoitf** - [PR #770](https://github.com/eclipse/che-docs/pull/770)
    - [x] Installing Che on Azure - **@benoitf** - [PR #769](https://github.com/eclipse/che-docs/pull/769)
- [x] Hosted Che - **@ibuziuk** - [PR #729](https://github.com/eclipse/che-docs/pull/729)
- [x] Che architecture **@l0rd** - [PR #756](https://github.com/eclipse/che-docs/pull/756/)
- [x] Che-Theia IDE basics - **@vparfonov** - [PR #763](https://github.com/eclipse/che-docs/pull/763)
- [x] Making a workspace portable using a Devfile - **@metlos** - [PR #752](https://github.com/eclipse/che-docs/pull/752)
- [x] Converting a Che 6 Workspace to a Che 7 devfile - **@olexii4** - [PR #766](https://github.com/eclipse/che-docs/pull/766)
- [x] Retrieving Che logs - **@tsmaeder** - [PR #765](https://github.com/eclipse/che-docs/pull/765)
- [x] Devfile and Plugin registries customization - **@l0rd** - [PR #767](https://github.com/eclipse/che-docs/pull/767)
- [x] Securing Che - **@skabashnyuk** - [PR #758](https://github.com/eclipse/che-docs/pull/758)
- [x] Creating a workspace from code sample - **@nickboldt** - [PR #737](https://github.com/eclipse/che-docs/pull/737)
- [x] Creating a workspace by importing source code of your project - **@nickboldt** - [PR #737](https://github.com/eclipse/che-docs/pull/737)
- [x] Importing a Kubernetes application into a Che workspace - **@l0rd** - [PR #760](https://github.com/eclipse/che-docs/pull/760)
- [x] Using a Visual Studio Code extension in Che - **@mikola** - [PR #754](https://github.com/eclipse/che-docs/pull/754/)
- [x] Remotely accessing your workspace (open a shell) - **@l0rd** - [PR #761](https://github.com/eclipse/che-docs/pull/761)
- [x] List the supported k8s resources in a devfile (c.f. https://github.com/eclipse/che/issues/12899) - **@slemeur** - [PR #753](https://github.com/eclipse/che-docs/pull/753)

#### Other documentation related tasks

- [x] Document known issue using github query **@rkratky**. Candidate issues:
      https://github.com/eclipse/che-operator/issues/31
      https://github.com/eclipse/che/issues/13607
      https://github.com/eclipse/che/issues/13442
- [x] Update issues template:
  - [x] for updating the content of bugs/questions/features https://github.com/eclipse/che/issues/13648 **@benoitf**
  - [x] main issues template like https://github.com/gatsbyjs/gatsby/issues/new/choose **@slemeur**
- [x] Update wiki and readme files **@slemeur** **@sunix**

## END 2 END TEST SUITE STATUS

Test suite description: https://github.com/eclipse/che/issues/12728

| STEP | IMPLEMENTED | STATUS | ISSUES|
| ---- | ------ | ----- | ----- |
| 1. Deployment (operator) | YES | NEED WORKAROUNDS | https://github.com/che-incubator/chectl/issues/92 https://github.com/che-incubator/chectl#145|
| 3. Start a Workspace from a devfile | YES | NEED WORKAROUNDS | https://github.com/eclipse/che/issues/13427  https://github.com/eclipse/che/issues/11867 https://github.com/eclipse/che/issues/13498 |
| 4. Use Java IDE features and the inner loop | PR: https://github.com/eclipse/che/pull/13615 | NOT TESTED | |

"Happy path" failures: 
- [x] Happy Path test is sometimes failing on "Error highlighting" step #13865 @rhopp
- [ ] Happy Path test sometimes fail on "Validation of debug functionality" #13864 @rhopp

## RELEASE ACCEPTANCE CRITERIA (manual tests)

N/A

## MARKETING AND PROMOTION (Postponed to September 2019)

Tasks to complete a couple of weeks after the release:

- [ ] Update Eclipse Che website @slemeur
  - [ ] Getting started factories updates @ibuziuk https://github.com/redhat-developer/rh-che/issues/1451 / https://github.com/redhat-developer/rh-che/issues/1490
- [ ] Blog post series of major Che 7 features (need a list) @slemeur
- [ ] Collect testimonials for Che 7 (user, OEM, plug-in developer) @slemeur
- [ ] Create sample tweets to use around Che 7 release for #chetips @jesse
- [ ] Prepare webinar to present Che 7 @slemeur @l0rd @gorkem

## FAQ

**What are the branches where the endgame issues/features will be merged? Master branch or a dedicated release branch?**
Master branch. A communication has been sent to che-dev to inform that for a couple of sprints we won't accept any PR that is not considered a blocker. For that purpose the following engineers will be responsible to approve (or not) PRs on the following repositories:
 
- [che server](https://github.com/eclipse/che): @l0rd, @rhopp, @nickboldt
- [devfile-registry](https://github.com/eclipse/che-devfile-registry): @l0rd, @rhopp, @skabashnyuk
- [plugin-registry](https://github.com/eclipse/che-devfile-registry): @l0rd, @rhopp, @skabashnyuk
- [che-theia](https://github.com/eclipse/che-theia): @l0rd, @rhopp, @evidolob
- [che-operator](https://github.com/eclipse/che-operator): @l0rd, @rhopp, @davidfestal
- [che-broker](https://github.com/eclipse/che-plugin-broker): @l0rd, @rhopp, @nickboldt
- [chectl](https://github.com/che-incubator/chectl): @l0rd, @rhopp, @benoitf

**Note that in any case PRs need to be reviewed by code owners as well (as usual)**

**How do we review new bugs and decide if they should be included to the endgame?**
A list of curators will do the triage of new issues (blocker/no blocker).
Issues that are not considered as blockers won't be included in the endgame.
The curator will send a message to the mattermost Eclipse Che chat to inform the community about the triaging.
These are the curators:
- Monday: @l0rd
- Tuesday: @nickboldt 
- Wednesday: @slemeur
- Thursday: @tsmaeder
- Friday: @rhopp

Issues that are not considered blockers but sev1 should be included in the 7.1 milestone

**How do we protect against last minute changes in 3rd party dependency (e.g. theia)? Are we going to create a branch/tag before the release?**
These are the tasks that we will take to use a stable version of Theia:
- Create a branch on theia repository (e.g. che-7.0.0) where we will cherry pick the commits that we want to include in Che @evidolob
- Create a branch on che-theia (e.g. 7.0.0) that should use theia branch mentioned above @evidolob
- Communicate to the Theia community @evidolob

**Are we continuing to do RC releases or are we going to wait until we have the endgame completed?**
We do releases as usual

**Comment by ibuziuk** [comment]
Can we remove `Hide issues on non che repositories (without migrating issues) @tsmaeder @evidolob @ibuziuk` since as I recall it was decided not to do this @slemeur @l0rd

**Comment by l0rd** [comment]
@ibuziuk done

**Comment by l0rd** [comment]
Che 7 has been released August 14 2019.
